tests/base.py

```diff
@@ -230,19 +230,22 @@ def expected_schema_keys(self, stream):
 
         return props.keys()
 
+    # TODO Determine if sorting is even a valid modifier for our expectations
```

Comment:
@luandy64 and I spent some time trying to avoid sorting. We modified in my PR the code to use assertDictEqual, but it still relies on sorting. We investigated using a library that can compare dicts but it threw errors at one point unfortunately. I really wish python or the community had a better solution to this since it's a very common problem people have to deal with. See https://github.com/singer-io/tap-square/pull/25/files#diff-19f36877654ab85cc34581bf5817a1abR190 - We aught to refactor the tests because I suspect there's a lot of repetition here.

Reply:
I used assertDictEqual in the other tests that are also comparing records line by line. This is very useful assertion.

Reply:
I have used Collections while debugging which is good when you have a suspected duplicate record.